packages/sdk-client-v3/src/client/client.ts

```diff
@@ -0,0 +1,187 @@
+    )
+
+  // Set default process options
+  const opt: ProcessOptions = {
+    total: Number.POSITIVE_INFINITY,
+    accumulate: true,
+    ...processOpt,
+  }
+
+  return new Promise((resolve: Function, reject: Function) => {
+    let _path: string,
+      _queryString = ''
+    if (request && request.uri) {
+      const [path, queryString] = request.uri.split('?')
+      _path = path
+      _queryString = queryString
+    }
+
+    const requestQuery = { ...qs.parse(_queryString) }
+    const query = {
+      // defaults
+      limit: 20,
+      // merge given query params
+      ...requestQuery,
+    }
+
+    let itemsToGet = opt.total
+    let hasFirstPageBeenProcessed = false
+    const processPage = async (lastId?: string, acc: Array<unknown> = []) => {
+      // Use the lesser value between limit and itemsToGet in query
+      const limit = query.limit < itemsToGet ? query.limit : itemsToGet
+      const originalQueryString = qs.stringify({ ...query, limit })
+
+      const enhancedQuery = {
+        sort: 'id asc',
+        withTotal: false,
+        ...(lastId ? { where: `id > "${lastId}"` } : {}),
+      }
+      const enhancedQueryString = qs.stringify(enhancedQuery)
+      const enhancedRequest = {
+        ...request,
+        uri: `${_path}?${enhancedQueryString}&${originalQueryString}`,
+      }
+
+      try {
+        const payload: ClientResult = await createClient(_options).execute(
+          enhancedRequest
+        )
+
+        const { results, count: resultsLength } = payload?.body || {}
+
+        if (!resultsLength && hasFirstPageBeenProcessed) {
+          return resolve(acc || [])
+        }
+
+        const result = await Promise.resolve(fn(payload))
+        // const result = await fn(payload)
+        let accumulated: Array<Omit<ClientResult, 'body'>>
+        hasFirstPageBeenProcessed = true
+
+        if (opt.accumulate) accumulated = acc.concat(result || [])
+
+        itemsToGet -= resultsLength
+        // If there are no more items to get, it means the total number
+        // of items in the original request have been fetched so we
+        // resolve the promise.
+        // Also, if we get less results in a page then the limit set it
+        // means that there are no more pages and that we can finally
+        // resolve the promise.
+        if (resultsLength < query.limit || !itemsToGet) {
+          return resolve(accumulated || [])
+        }
+
+        const last = results[resultsLength - 1]
+        const newLastId = last && last.id
+
+        processPage(newLastId, accumulated)
+      } catch (error) {
+        reject(error)
+        // return {
+        //   error
+        // }
+      }
+    }
+
+    // Start iterating through pages
+    processPage()
+  })
+}
+
+export default function createClient(middlewares: ClientOptions): Client {
+  _options = middlewares
+  validateClient(middlewares)
+
+  const resolver = {
+    async resolve(rs: ClientRequest): Promise<ClientResult> {
```

Comment:
I think the variable name here is fine.

Reply:
this `rs` variable from my point of view is not clear enough. And as I can see you are calling it in the rest of the function.
I think it could be better explained